pennylane/templates/subroutines/aqft.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the template for AQFT.
+"""
+# pylint:disable=abstract-method,arguments-differ,protected-access
+
+import warnings
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation
+
+
+class AQFT(Operation):
+    r"""stuff"""
+    num_wires = AnyWires
+    grad_method = None
+
+    def __init__(self, order, wires=None, id=None):
+        wires = qml.wires.Wires(wires)
+
+        self.hyperparameters["n_wires"] = len(wires)
+
+        if not isinstance(order, int):
+            warnings.warn(f"The order must be an integer. Using order = {round(order)}")
+            order = round(order)
+
+        if order >= self.hyperparameters["n_wires"] - 1:
+            warnings.warn(
+                f'The order ({order}) is >= to the number of wires - 1 ({self.hyperparameters["n_wires"]-1}). Using the QFT class is recommended in this case.'
+            )
+            order = self.hyperparameters["n_wires"] - 1
+
+        if order < 0:
+            raise ValueError("Order can not be less than 0")
+
+        if order == 0:
+            warnings.warn("order=0, applying Hadamard transform")
+
+        self.hyperparameters["order"] = order
+        super().__init__(wires=wires, id=id)
+
+    @property
+    def num_params(self):
+        return 0
+
+    @staticmethod
+    def compute_decomposition(
+        wires, n_wires, order
```

Comment:
I am following the template from QFT, which has `n_wires`

Reply:
If you delete it, it still works? I don't think it is necessary 🤔

Reply:
will not work if I delete it simply from here as `n_wires` is being initialized as a hyperparameter and thus this function expects it here.
It can be removed if I remove it from the hyperparameters directly. 
Should I do that?

Reply:
Let's do that, we can remove it from both places :)